src/tasks/LibraryBuilder/Templates/load_assemblies.c

```diff
@@ -1,9 +1,27 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
+#include <assert.h>
+#include <string.h>
+
 #include <mono/metadata/assembly.h>
 
 MonoAssembly* mono_assembly_open (const char *filename, MonoImageOpenStatus *status);
 
-void load_assemblies_with_exported_symbols ()
+void
+mono_assembly_open_from_dir (const char *dir, const char* filename)
+{
+    int str_len = strlen (dir) + strlen (filename) + 2;
+    const char *assembly_path = (char*)malloc (sizeof (char) * str_len);
+    int num_char = snprintf (assembly_path, str_len, "%s/%s", dir, filename);
+
+    assert (num_char > 0 && num_char < str_len);
+
+    mono_assembly_open (assembly_path, NULL);
```

Comment:
oh, @mdh1418 I realized I was wrong yesterday - `mono_assembly_open` works on assembly file paths; `mono_assembly_load` (which takes a `MonoAssemblyName*`) works with assembly names.

So the fact that you use `mono_set_assemblies_path` won't help `mono_assembly_open`, but it will help `mono_assembly_load` (and the runtime internals that also do assembly probing by paths).

Reply:
Ah, will `mono_assembly_load` also load the GOT slots? I'm not familiar with the differences yet, will have to look.
It also looks like `mono_assembly_load` takes in a basedir, but not sure how it gets used quite yet, maybe `mono_assembly_load_with_partial_name` might do the trick, will have to try it out.

Reply:
`mono_assembly_load_with_partial_name` worked!